src/coreclr/tools/Common/TypeSystem/IL/ILStackHelper.cs

```diff
@@ -436,38 +400,18 @@ public static int ComputeMaxStack(this MethodIL methodIL)
                         Debug.Assert(stackHeight > 0);
                         break;
 
-                    case ILOpcode.switch_:
-                        Debug.Assert(stackHeight > 0);
-                        isVariableSize = true;
-                        stackHeight -= 1;
-                        currentOffset += 1 + (ReadInt32(ilbytes, currentOffset + 1) * 4) + 4;
-                        break;
-
                     default:
                         Debug.Fail("Unknown instruction");
                         break;
                 }
 
-                if (!isVariableSize)
-                    currentOffset += opcode.GetSize();
+                if (!hasReadOperand)
```

Comment:
Nit: wasn't `isVariableSize` better? We defaulted to false and had to set to true in one case. Now we still default to false, but need to set it to true in two cases.

Reply:
I thought about name of this for a whiel. I think the meaning of the variable changed because it's using `ILReader` now.
Before we manually kept track of the `currentOffset` and `isVariableSize` was used to skip the `ILOpcode.switch_` case where the `currentOffset` was calculated manually. 
Now by using `ILReader.Skip(opcode)` (which essentially is `operandSize - op.GetSize()` + switch handling) we can avoid this manual tracking, however because we want to read the operand in two cases for stack height tracking purposes (branches and calls/newobj) we need to flip the bit to avoid "reading" the operand twice in `ILReader`. That was why I called it `hasReadOperand`. 

tldr: the current branches that use the bool are not variable sized, switch was the only special case and we now handle it in `ILReader.Skip(op)`